lib/broccoli/ember-app.js

```diff
@@ -1341,15 +1321,17 @@ EmberApp.prototype.styles = function() {
     }.bind(this));
   }
 
-  this.import('vendor/addons.css');
-
   var vendorStyles = [];
   for (var outputFile in this._styleOutputFiles) {
+    var isMainVendorFile = outputFile === this.options.outputPaths.vendor.css;
     var headerFiles = this._styleOutputFiles[outputFile];
+    var inputFiles = isMainVendorFile ? ['modules/**/*.css'] : [];
```

Comment:
> Also, anybody overrides treeForAddon (public!) and all bets are off.

We were already only concatting JS from `modules/` dir. `treeForAddon` is expected to return a tree with that directory (which is another thing that I dislike and want to kill, but takes time).  So we are definitely safe here WRT `*.js` assets. 

> Pulls assets from anywhere inside of the treeForAddon results.

Agreed, this is a great point (for those following along at home, @nathanhammond just correctly pointed out that we were only concatting `*.js` files from `modules/` but we would concat `*.css` files from anywhere in the tree returned by `treeForAddon`). I will address shortly.